lib/lhm/chunker.rb

```diff
@@ -16,7 +16,9 @@ class Chunker
     def initialize(migration, connection = nil, options = {})
       @migration = migration
       @connection = connection
-      @throttler = options[:throttler]
+      if @throttler = options[:throttler]
+        @throttler.connection = @connection if @throttler.respond_to?(:connection=)
```

Comment:
I had a really hard time figuring out what to do with this in the initial PR. The only existing Throttler has no required parameters and it became a little awkward to get this to work. I'm not very happy with how it ended up.

I don't like this way much more and agree with @camilo -- this creates a weird implicit contract on the SlaveLag constructor. The original isn't much better but at least it blew up ASAP when you didn't provide a connection.

> So... what about instead of passing instances around we pass the throttler class as one arg and the throttler construcrtor args as another argument...

That might work better but it's definitely a larger refactor. Not huge, but it will have a wider impact.